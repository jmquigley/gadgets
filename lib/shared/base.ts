import * as React from 'react';
import {Size} from './index';

const styles = require('./styles.css');

export interface BaseOptions {
	className?: boolean;
	disabled?: boolean;
	visible?: boolean;
}

const defaultBaseOptions: BaseOptions = {
	className: true,
	disabled: true,
	visible: true
};

/**
 * A base class between React and all components in the module.
 */
export abstract class BaseComponent<P, S> extends React.Component<P, S> {

	protected _classes: string[] = [];
	protected _inlineStyle: any = {};    // inline style overrides
	protected _styles: any = {};         // css modules styles

	constructor(props: P, pstyles: any = {}) {
		super(props);
		this._styles = pstyles;
	}

	get classes(): string[] {
		return this._classes;
	}

	set classes(str: string[]) {
		this._classes = str;
	}

	get inlineStyle(): any {
		return this._inlineStyle;
	}

	set inlineStyle(val: any) {
		this._inlineStyle = Object.assign(this._inlineStyle, val);
	}

	get style(): string {
		return this._inlineStyle;
	}

	get styles(): any {
		return this._styles;
	}

	protected buildStyles(props: P, style: any = {}, opts?: BaseOptions): void {
		this._classes = [];
		this._inlineStyle = Object.assign(this._inlineStyle, props['style'], style);

		opts = Object.assign(
			defaultBaseOptions,
			opts
		);

		if (props['className'] !== '' && opts.className) {
			this._classes.push(props['className']);
		}

		if (!props['visible'] && opts.visible) {
			this._classes.push(this.styles.invisible);
		}

		if (props['disabled'] && opts.disabled) {
			this._classes.push(styles.disabled);
			this._classes.push('nohover');
		}
	}

	/**
	 * Takes a sizing parameter and a styles object and searches for the
	 * corresponding style class CSS.  If it is found, then it returns the local
	 * style for that size.
	 * @returns {string} the name of the local style for that size.  If
	 * it is not found, then an empty string is returned.
	 */
	protected getSizeStyle() {

		switch (this.props['size']) {
		case Size.xxsmall:
			if ('xxsmall' in this.styles) {
				return this.styles.xxsmall;
			}
			break;

		case Size.xsmall:
			if ('xsmall' in this.styles) {
				return this.styles.xsmall;
			}
			break;

		case Size.small:
			if ('small' in this.styles) {
				return this.styles.small;
			}
			break;

		case Size.large:
			if ('large' in this.styles) {
				return this.styles.large;
			}
			break;

		case Size.xlarge:
			if ('xlarge' in this.styles) {
				return this.styles.xlarge;
			}
			break;

		case Size.xxlarge:
			if ('xxlarge' in this.styles) {
				return this.styles.xxlarge;
			}
			break;

		case Size.normal:
		case Size.medium:
		default:
			if ('medium' in this.styles) {
				return this.styles.medium;
			}
		}

		return '';
	}
}
