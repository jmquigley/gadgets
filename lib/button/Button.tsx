'use strict';

import {cloneDeep} from 'lodash';
import * as React from 'react';
import {baseClasses, BaseProps, getDefaultBaseProps} from '../shared';

const styles = require('./styles.css');

export interface ButtonProps extends BaseProps {
	iconName?: string;      // font awesome string
	style?: any;
}

export function getDefaultButtonProps(): ButtonProps {
	return cloneDeep(Object.assign(getDefaultBaseProps(), {
		iconName: 'bomb',
		style: {}
	}));
}

export const ButtonComponent = (props: ButtonProps) => (
    <i
        className={props.classes.join(' ')}
        onClick={props.onClick}
		aria-hidden="true"
	    disabled={props.disabled}
		style={props.style}
	>
	</i>
);

/**
 * A typical button control widget.  This control only uses an icon and no text to represent the button.
 * The icons are [Font Awesome](http://fontawesome.io/) strings.  That library is built into this module,
 * so any font available in the current release of that library is available.
 *
 * #### Examples:
 *
 * ```javascript
 * import {Button} from 'gadgets';
 * <Button iconName="cab" onClick={someFunction} />
 * ```
 *
 * #### Events
 * - `onClick`` - when the button control is clicked by the user
 *
 * #### Styles
 * - `ui-button` - top level style placed on `<i>` control that constructs the button.
 *
 * #### Properties
 * - `classes: string[]` - an array of strings that represent custom CSS class names that will be applied
 * to the component.
 * - `disabled: boolean` - if true then then the control is disabled.  Defaults to false.
 * - iconName: string` - the name of the font awesome icon used with this button
 * - `noripple: boolean` - turns off the ripple effect for a button.  On by default.
 * - `style: object` - an object of key value/pairs that represent CSS style settings
 * - `visible: boolean` - turns the display of this control on or off.  If true, then the component can
 * be seen.  The default is true.
 *
 */
export class Button extends React.Component<ButtonProps, undefined> {

    public static defaultProps: ButtonProps = getDefaultButtonProps();

    constructor(props: ButtonProps) {
		super(props);
	}

	private buildClasses = () => {
		let l: string[] = baseClasses(this.props);

		l.push(styles.button);
		l.push('fa');
		l.push(`fa-${this.props.iconName}`);
		l.push('ui-button');

		if (!this.props.noripple && !this.props.disabled) {
			l.push('ripple');
		}

		return l;
	}

	private handleClick = (e: Event) => {
		if (!this.props.disabled && this.props.visible && this.props.onClick != null) {
			this.props.onClick();
		}
		e.stopPropagation();
	}

	render() {
		return (
			<ButtonComponent
				{...this.props}
				onClick={this.handleClick}
				classes={this.buildClasses()}
			/>
		);
	}
}
