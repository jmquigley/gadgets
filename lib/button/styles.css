:local {
	.button {
		text-align: center;
		vertical-align: bottom;
		outline: none;
		width: 32px;
		background-color: inherit;
		line-height: inherit;
	}

	.button:not(.nohover):hover {
		background-color: lightgray;
	}

	.buttonDisabled {
		opacity: 0.5;
	}

	.buttonInvisible {
		display: none;
		width: 0;
	}
}
