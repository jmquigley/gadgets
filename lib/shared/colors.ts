export type ColorType = string;

export enum ColorScheme {
	c1 = "#004358",
	c2 = "#1f8a70",
	c3 = "#bedb39",
	c4 = "#ffe11a",
	c5 = "#fd7400"
}

export enum Color {
	black = "#000000",
	error = "#d9534f",
	glow = "#51cbee",
	gray = "#666666",
	info = "#5bc0de",
	lightgray = "#cccccc",
	lightsaber = "#2ff923",
	red = "#ff0000",
	silver = "#c0c0c0",
	slategray = "#708090",
	success = "#5cb85c",
	ultralightgray = "#e6e6e6",
	warning = "#f0ad4e",
	white = "#ffffff"
}
