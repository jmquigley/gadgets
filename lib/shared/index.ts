import {
	BaseComponent,
	baseZIndex,
	defaultSize
} from './base';

import {
	getDefaultWrapperProps,
	Wrapper,
	WrapperProps
} from './wrapper';

import {
	BaseProps,
	Color,
	ColorScheme,
	Direction,
	disabled,
	DisabledCSS,
	getDefaultBaseProps,
	invisible,
	InvisibleCSS,
	Justify,
	Location,
	locationStyle,
	SortOrder,
	Styles
} from './props';

import {
	borderStyle,
	boxStyle,
	fontStyle,
	FontStyle,
	lineHeightStyle,
	rectStyle,
	Sizes,
	Sizing,
	Styling
} from './sizing';

import {
	BaseState,
	getDefaultBaseState
} from './state';

import {
	getTheme,
	getThemeList,
	setTheme,
	Theme,
	ThemeProps
} from './themes';

export {
	BaseComponent,
	BaseProps,
	BaseState,
	baseZIndex,
	borderStyle,
	boxStyle,
	Color,
	ColorScheme,
	defaultSize,
	Direction,
	disabled,
	DisabledCSS,
	fontStyle,
	FontStyle,
	getDefaultBaseProps,
	getDefaultBaseState,
	getDefaultWrapperProps,
	getTheme,
	getThemeList,
	invisible,
	InvisibleCSS,
	Justify,
	lineHeightStyle,
	Location,
	locationStyle,
	rectStyle,
	setTheme,
	Sizes,
	Sizing,
	SortOrder,
	Styles,
	Styling,
	Theme,
	ThemeProps,
	Wrapper,
	WrapperProps
};
