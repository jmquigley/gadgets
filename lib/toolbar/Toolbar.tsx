/**
 * A typical toolbar control.  It takes a set of buttons and controls and
 * displays the on a horizontal control.  It will only accept a specific
 * set of controls (from the Gadgets library)
 *
 * - Button
 * - ButtonCircle
 * - ButtonDialog
 * - ButtonText
 * - ButtonToggle
 * - Divider
 * - Dropdown
 * - Option
 * - Switch
 * - TextField
 *
 * ## Screen:
 * <img src="https://github.com/jmquigley/gadgets/blob/master/images/toolbar.png" width="60%" />
 *
 * ## Examples:
 *
 * ```javascript
 * import {Button, Sizing, Toolbar} from 'gadgets';
 *
 * <Toolbar justify={Justify.left} sizing={Sizing.small}>
 *     <Button iconName="cab" onClick={someFunction} />
 *     <Divider />
 *     <Option />
 *     ...
 * </Toolbar>
 * ```
 *
 * ## API
 * #### Events
 * None
 *
 * #### Styles
 * - `ui-toolbar` - global style placed on the root `<div>` component of the
 * toolbar.
 * - `ui-toolbar-group` - The components are all placed within a grouping
 * div to set its left/right/center justification.  This global style is
 * placed on this div.
 *
 * #### Properties
 * - `justify: {Justify} (Justify.left)` - The toolbar can be placed to the left
 * (default), center, or right within its container.  The property sets that
 * location.
 *
 * @module Toolbar
 */

'use strict';

// const debug = require('debug')('Toolbar');

import {cloneDeep} from 'lodash';
import * as React from 'react';
import {BinaryTree} from 'util.ds';
import {Keys} from 'util.keys';
import {
	BaseComponent,
	BaseProps,
	getDefaultBaseProps,
	Justify,
	Wrapper
} from '../shared';
import styled, {withProps} from '../shared/themed-components';

export interface ToolbarProps extends BaseProps {
	justify?: Justify;
}

export function getDefaultToolbarProps(): ToolbarProps {
	return cloneDeep(Object.assign({},
		getDefaultBaseProps(), {
			justify: Justify.left,
			obj: 'Toolbar'
		})
	);
}

export const ToolbarView: any = withProps<ToolbarProps, HTMLDivElement>(styled.div)`
	border: solid 1px silver;
	box-sizing: border-box;
	display: flex;
	padding: 3px 2px;
`;

export const ToolbarGroupView: any = withProps<ToolbarProps, HTMLDivElement>(styled.div)`
	align-items: center;
	display: flex;
	padding: 2px 0 1px 0;

	${props => {
		switch (props.justify) {
			case Justify.center: return ('margin: auto;');
			case Justify.right: return ('margin-left: auto;');
			default: return('');
		}
	}}
`;

export const ToolbarElementView: any = withProps<ToolbarProps, HTMLDivElement>(styled.div)`
	box-sizing: border-box;
`;

export class Toolbar extends BaseComponent<ToolbarProps, undefined> {

	public static readonly defaultProps: ToolbarProps = getDefaultToolbarProps();

	private _keys: Keys;
	private static readonly _whitelist = new BinaryTree([
		'Button',
		'ButtonCircle',
		'ButtonDialog',
		'ButtonText',
		'ButtonToggle',
		'Divider',
		'Dropdown',
		'Option',
		'StyledComponent',
		'Switch',
		'TextField'
	]);

	constructor(props: ToolbarProps) {
		super(props, Toolbar.defaultProps.style);

		this._keys = new Keys({testing: this.props.testing});
		this._classes.add('ui-toolbar');

		this.componentWillUpdate(this.props);
	}

	public render() {
		const components: any = [];

		React.Children.forEach(this.props.children, (child: any, idx: number) => {
			if (Toolbar._whitelist.contains(child['props'].obj)) {
				const style = Object.assign({}, child['props'].style, {
					display: 'flex',
					height: this.fontSizePX(this.props.sizing, 1.5),
					margin: '0 2px'
				});

				switch (child['props'].obj) {
					case 'Button':
					case 'ButtonCircle':
					case 'ButtonDialog':
					case 'ButtonToggle':
						style['width'] = this.fontSizePX(this.props.sizing, 1.5);

					case 'ButtonText':
						style['border'] = `solid 1px ${this.theme.borderColor}`,
						delete style['width'];
						break;

					case 'Switch':
						style['padding-top'] = '0.1em';
						style['margin'] = '0 6px';
						break;

					case 'TextField':
					case 'StyledComponent':
						delete style['height'];
						break;
				}

				if (child['props'].obj === 'ButtonCircle') {
					delete style['border'];
				}

				const newChild = React.cloneElement(child as any, {
					className: 'ui-toolbar-element',
					disabled: this.props.disabled,
					sizing: this.props.sizing,
					visible: this.props.visible
				});

				components.push(
					<ToolbarElementView
						key={this._keys.at(idx)}
						style={style}
					>
						{newChild}
					</ToolbarElementView>
				);
			}
		});

		return(
			<Wrapper {...this.props} >
				<ToolbarView className={this.classes}>
					<ToolbarGroupView
						className="ui-toolbar-group"
						justify={this.props.justify}
					>
						{components}
					</ToolbarGroupView>
				</ToolbarView>
			</Wrapper>
		);
	}
}
