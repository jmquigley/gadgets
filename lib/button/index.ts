import {IconSize} from '../icon';
import {Button, ButtonProps, getDefaultButtonProps} from './Button';

export {
	Button,
	ButtonProps,
	getDefaultButtonProps,
	IconSize
};
