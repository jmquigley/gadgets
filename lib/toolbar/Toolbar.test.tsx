import {shallow} from "enzyme";
import assert from "power-assert";
import * as React from "react";
import {Button} from "../button/Button";
import {Justify, Sizing} from "../shared";
import Toolbar from "./Toolbar";

[Justify.left, Justify.right, Justify.center].forEach((justify: Justify) => {
	test(`Test creation of a Toolbar instance with ${Justify[justify]} justification`, () => {
		const ctl = shallow(
			<Toolbar justify={justify} sizing={Sizing.small}>
				<Button iconName='car' />
				<div>blah</div>
			</Toolbar>
		);

		assert(ctl);
		expect(ctl).toMatchSnapshot();
	});
});

test("Test disabling of a Toolbar instance", () => {
	const ctl = shallow(
		<Toolbar disabled>
			<Button iconName='car' />
			<div>blah</div>
		</Toolbar>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test("Test making the Toolbar invisible", () => {
	const ctl = shallow(
		<Toolbar visible={false}>
			<Button iconName='car' />
			<div>blah</div>
		</Toolbar>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});
