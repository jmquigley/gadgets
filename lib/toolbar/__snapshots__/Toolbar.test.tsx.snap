// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a Toolbar instance with center justification 1`] = `
<Wrapper
  bottom={null}
  className={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  justify={2}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={false}
  notheme={false}
  obj="Toolbar"
  onError={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="small"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <styled.div
    className="ui-toolbar"
  >
    <styled.div
      className="ui-toolbar-group"
      justify={2}
    >
      <styled.div
        key="0"
        style={
          Object {
            "border": "solid 1px #c0c0c0",
            "display": "flex",
            "height": "18px",
            "margin": "0 2px",
          }
        }
      >
        <Button
          bottom={null}
          className="ui-toolbar-element"
          contentEditable={false}
          controlled={true}
          direction="right"
          disabled={false}
          err={null}
          errorMessage={null}
          focus={false}
          height={null}
          iconName="car"
          iconStyle=""
          id={null}
          left={null}
          location="none"
          minHeight={null}
          minWidth={null}
          noedit={false}
          nohover={false}
          noripple={false}
          notheme={false}
          obj="Button"
          onClick={[Function]}
          padding={null}
          selected={false}
          sizing="small"
          style={Object {}}
          testing={true}
          theme={null}
          tooltip={null}
          top={null}
          visible={true}
          width={null}
          xcss={null}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</Wrapper>
`;

exports[`Test creation of a Toolbar instance with left justification 1`] = `
<Wrapper
  bottom={null}
  className={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  justify={0}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={false}
  notheme={false}
  obj="Toolbar"
  onError={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="small"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <styled.div
    className="ui-toolbar"
  >
    <styled.div
      className="ui-toolbar-group"
      justify={0}
    >
      <styled.div
        key="0"
        style={
          Object {
            "border": "solid 1px #c0c0c0",
            "display": "flex",
            "height": "18px",
            "margin": "0 2px",
          }
        }
      >
        <Button
          bottom={null}
          className="ui-toolbar-element"
          contentEditable={false}
          controlled={true}
          direction="right"
          disabled={false}
          err={null}
          errorMessage={null}
          focus={false}
          height={null}
          iconName="car"
          iconStyle=""
          id={null}
          left={null}
          location="none"
          minHeight={null}
          minWidth={null}
          noedit={false}
          nohover={false}
          noripple={false}
          notheme={false}
          obj="Button"
          onClick={[Function]}
          padding={null}
          selected={false}
          sizing="small"
          style={Object {}}
          testing={true}
          theme={null}
          tooltip={null}
          top={null}
          visible={true}
          width={null}
          xcss={null}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</Wrapper>
`;

exports[`Test creation of a Toolbar instance with right justification 1`] = `
<Wrapper
  bottom={null}
  className={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  justify={1}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={false}
  notheme={false}
  obj="Toolbar"
  onError={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="small"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <styled.div
    className="ui-toolbar"
  >
    <styled.div
      className="ui-toolbar-group"
      justify={1}
    >
      <styled.div
        key="0"
        style={
          Object {
            "border": "solid 1px #c0c0c0",
            "display": "flex",
            "height": "18px",
            "margin": "0 2px",
          }
        }
      >
        <Button
          bottom={null}
          className="ui-toolbar-element"
          contentEditable={false}
          controlled={true}
          direction="right"
          disabled={false}
          err={null}
          errorMessage={null}
          focus={false}
          height={null}
          iconName="car"
          iconStyle=""
          id={null}
          left={null}
          location="none"
          minHeight={null}
          minWidth={null}
          noedit={false}
          nohover={false}
          noripple={false}
          notheme={false}
          obj="Button"
          onClick={[Function]}
          padding={null}
          selected={false}
          sizing="small"
          style={Object {}}
          testing={true}
          theme={null}
          tooltip={null}
          top={null}
          visible={true}
          width={null}
          xcss={null}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</Wrapper>
`;

exports[`Test disabling of a Toolbar instance 1`] = `
<Wrapper
  bottom={null}
  className={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  justify={0}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={false}
  notheme={false}
  obj="Toolbar"
  onError={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <styled.div
    className="ui-toolbar"
  >
    <styled.div
      className="ui-toolbar-group"
      justify={0}
    >
      <styled.div
        key="0"
        style={
          Object {
            "border": "solid 1px #c0c0c0",
            "display": "flex",
            "height": "24px",
            "margin": "0 2px",
          }
        }
      >
        <Button
          bottom={null}
          className="ui-toolbar-element"
          contentEditable={false}
          controlled={true}
          direction="right"
          disabled={true}
          err={null}
          errorMessage={null}
          focus={false}
          height={null}
          iconName="car"
          iconStyle=""
          id={null}
          left={null}
          location="none"
          minHeight={null}
          minWidth={null}
          noedit={false}
          nohover={false}
          noripple={false}
          notheme={false}
          obj="Button"
          onClick={[Function]}
          padding={null}
          selected={false}
          sizing="normal"
          style={Object {}}
          testing={true}
          theme={null}
          tooltip={null}
          top={null}
          visible={true}
          width={null}
          xcss={null}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</Wrapper>
`;

exports[`Test making the Toolbar invisible 1`] = `
<Wrapper
  bottom={null}
  className={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  justify={0}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={false}
  notheme={false}
  obj="Toolbar"
  onError={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={false}
  width={null}
  xcss={null}
>
  <styled.div
    className="ui-toolbar"
  >
    <styled.div
      className="ui-toolbar-group"
      justify={0}
    >
      <styled.div
        key="0"
        style={
          Object {
            "border": "solid 1px #c0c0c0",
            "display": "flex",
            "height": "24px",
            "margin": "0 2px",
          }
        }
      >
        <Button
          bottom={null}
          className="ui-toolbar-element"
          contentEditable={false}
          controlled={true}
          direction="right"
          disabled={false}
          err={null}
          errorMessage={null}
          focus={false}
          height={null}
          iconName="car"
          iconStyle=""
          id={null}
          left={null}
          location="none"
          minHeight={null}
          minWidth={null}
          noedit={false}
          nohover={false}
          noripple={false}
          notheme={false}
          obj="Button"
          onClick={[Function]}
          padding={null}
          selected={false}
          sizing="normal"
          style={Object {}}
          testing={true}
          theme={null}
          tooltip={null}
          top={null}
          visible={false}
          width={null}
          xcss={null}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</Wrapper>
`;

exports[`Test retrieval of Toolbar props object 1`] = `
Object {
  "bottom": null,
  "children": null,
  "className": null,
  "contentEditable": false,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "id": null,
  "justify": 0,
  "left": null,
  "location": "none",
  "minHeight": null,
  "minWidth": null,
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "notheme": false,
  "obj": "Toolbar",
  "padding": null,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;
